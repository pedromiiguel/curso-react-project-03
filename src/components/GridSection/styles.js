/* eslint-disable no-unused-vars */
import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    /* grid-template-columns: repeat(auto-fill, minmax(280px, 1fr)); */
    grid-template-columns: 1fr 1fr 1fr;

    gap: ${theme.spacings.large};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      width: 80%;
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -2rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
