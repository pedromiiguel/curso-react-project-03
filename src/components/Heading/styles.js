import styled, { css } from 'styled-components';

export const Title = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
  `}
`;
