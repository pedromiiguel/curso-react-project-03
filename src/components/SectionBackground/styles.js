import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${background ? theme.colors.primaryColor : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.primaryColor};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
