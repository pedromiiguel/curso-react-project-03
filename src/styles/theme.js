export const theme = {
  colors: {
    primaryColor: '#0a1128',
    secondaryColor: '#DC143C',
    lightBg: '#ffffff',
    darkBg: '#0A1128',
  },

  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
};
