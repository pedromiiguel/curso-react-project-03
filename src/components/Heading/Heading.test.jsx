import { screen } from '@testing-library/dom';
import { ThemeProvider } from 'styled-components';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.fonts.sizes.small,
      'text-transform': 'none',
    });
  });

  it('should render with white color', () => {
    renderTheme(<Heading colorDark={false}>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.lightBg,
    });
  });

  it('should render correct heading sizes', () => {
    const { rerender } = renderTheme(<Heading size="small">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.fonts.sizes.small,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="medium">Texto</Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'Texto' })).toHaveStyle({
      'font-size': theme.fonts.sizes.large,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="big">Texto</Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'Texto' })).toHaveStyle({
      'font-size': theme.fonts.sizes.xlarge,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="huge">Texto</Heading>
      </ThemeProvider>,
    );
    expect(screen.getByRole('heading', { name: 'Texto' })).toHaveStyle({
      'font-size': theme.fonts.sizes.xhuge,
    });
  });

  it('should render correct font-size when using mobile', () => {
    renderTheme(<Heading size="huge">texto</Heading>);
    screen.getByRole('heading', { name: 'texto' });

    expect(screen.getByRole('heading', { name: 'texto' })).toHaveStyleRule(
      'font-size',
      theme.fonts.sizes.xlarge,
      {
        media: theme.media.lteMedium,
      },
    );
  });

  it('should render with uppercase letters', () => {
    renderTheme(<Heading uppercase>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });
    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6">Texto</Heading>);
    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading as="h6">Texto</Heading>);
    expect(container).toMatchSnapshot();
  });
});
