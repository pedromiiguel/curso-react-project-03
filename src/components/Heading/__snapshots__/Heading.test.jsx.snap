// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  font-size: 1.6rem;
  text-transform: none;
}

<div>
  <h6
    class="c0"
  >
    Texto
  </h6>
</div>
`;
